src/query/graphite/common/percentiles.go

```diff
@@ -71,6 +71,53 @@ func SafeSort(input []float64) int {
 	return nans
 }
 
+// SafeSum returns the sum of the input slice the number of NaNs in the input.
+func SafeSum(input []float64) (float64, int) {
+	nans := 0
+	sum := 0.0
+	for _, v := range input {
+		if !math.IsNaN(v) {
+			sum += v
+		} else {
+			nans += 1
+		}
+	}
+	return sum, nans
+}
+
+// SafeMax returns the maximum value of the input slice the number of NaNs in the input.
+func SafeMax(input []float64) (float64, int) {
+	nans := 0
+	max := -math.MaxFloat64
+	for _, v := range input {
+		if !math.IsNaN(v) {
+		if v > max {
+			max = v
+		}
+		} else {
+			nans += 1
+		}
+	}
+	return max, nans
+}
+
+// SafeMin returns the minimum value of the input slice the number of NaNs in the input.
+func SafeMin(input []float64) (float64, int) {
+	nans := 0
+	min := math.MaxFloat64
+	for _, v := range input {
+		if !math.IsNaN(v) {
+			if v < min {
```

Comment:
nit: Go prefers to limit indentation wherever possible (since it's relative to complexity of the statement). Can simplify loop by following's advice of Effective Go "In the Go libraries, you'll find that when an if statement doesn't flow into the next statement—that is, the body ends in break, continue, goto, or return—the unnecessary else is omitted.":
https://golang.org/doc/effective_go.html#control-structures

e.g.
```go
for _, v := range input {
  if math.IsNaN(v) {
    nans++
    continue
  }
  if v < min {
    min = v
  }
}
```

Reply:
done!